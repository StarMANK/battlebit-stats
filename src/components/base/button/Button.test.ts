import { mount } from '@vue/test-utils'
import Component from './Button.vue'

describe('Button.vue', () => {
  it('should render', () => {
    const wrapper = mount(Component)
    // const wrapper = mount(Component, { props: { initial: 10 } })
    expect(wrapper.text()).toContain('click me')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should show loading spinner', () => {
    const wrapper = mount(Component, { propsData: { isLoading: true } })

    // const wrapper = mount(Component, { props: { initial: 10 } })
    expect(wrapper.find('[data-testid="loadingSpinner"]')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  // it('should be interactive', async() => {
  //   const wrapper = mount(Counter, { props: { initial: 0 } })
  //   expect(wrapper.text()).toContain('0')

  //   expect(wrapper.find('.inc').exists()).toBe(true)

  //   await wrapper.get('button').trigger('click')

  //   expect(wrapper.text()).toContain('1')
  // })
})
