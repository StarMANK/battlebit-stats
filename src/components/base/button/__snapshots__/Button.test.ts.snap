// Vitest Snapshot v1

exports[`Button.vue > should render 1`] = `
"<button class=\\"button inline-flex default md text-white\\" type=\\"button\\" data-v-4cf7dfa1=\\"\\">
  <!-- @slot Custom loading content, otherwise a spinner. Loading replaces prefix -->
  <!--v-if-->
  <!-- @slot Used for the button label --><span class=\\"capitalize-first-letter\\" data-v-4cf7dfa1=\\"\\">click me</span>
  <!--v-if-->
</button>"
`;

exports[`Button.vue > should show loading spinner 1`] = `
"<button class=\\"button inline-flex default md loading disabled text-white\\" type=\\"button\\" disabled=\\"\\" data-v-4cf7dfa1=\\"\\">
  <!-- @slot Custom loading content, otherwise a spinner. Loading replaces prefix -->
  <fontawesomeicon icon=\\"[object Object]\\" spin=\\"\\" fixed-width=\\"\\" data-testid=\\"loadingSpinner\\" class=\\"mr-1\\" data-v-4cf7dfa1=\\"\\"></fontawesomeicon><!-- @slot Used for the button label --><span class=\\"capitalize-first-letter\\" data-v-4cf7dfa1=\\"\\">click me</span>
  <!--v-if-->
</button>"
`;
